pkg/token/execCredentialPlugin.go

```diff
@@ -47,7 +47,7 @@ func (p *execCredentialPlugin) Do() error {
 	}
 
 	// verify resource
-	if token.Resource == p.o.ServerID && !token.IsZero() {
+	if (token.Resource == p.o.ServerID || token.Resource == fmt.Sprintf("%s:%s", servicePrincipalLogin, p.o.ServerID)) && !token.IsZero() {
```

Comment:
```suggestion
	targetAudience := p.o.ServerID
	if p.o.IsLegacy {
		targetAudience = fmt.Sprintf("spn:%s", p.o.ServerID)
	}
	if token.Resource == targetAudience && !token.IsZero() {
```
since we know whether it's legacy or not, lets do a proper validation based on legacy flag